### PR TITLE
psh: tests improvements

### PR DESCRIPTION
**Pull request by mateusz-bloch**
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [x] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [ ] Tested by hand on: (list targets here).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [x] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [x] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: psh/test-cat.py [hunk 1]
```diff
@@ -27,8 +25,61 @@ def assert_cat_h(p):
...
 
 
+def assert_cat_err(p):
+    fname = 'nonexistentFile'
+
+    statement = f'cat: {fname}: ENOENT'
+    psh.assert_cmd(p, f'cat {fname}', expected=statement, result='fail')
+
+    statement = rf'cat: {ROOT_TEST_DIR}: EISDIR'
+    psh.assert_cmd(p, f'cat {ROOT_TEST_DIR}', expected=statement, result='fail')
```
Comment (damianloew) [inline psh/test-cat.py:35]: We want to leave it flexible (for example on linux the behavior is a little but different: `cat: nonexistent: No such file or directory` - maybe we will change the approach in the future). However checking cat on directory is a good idea and we can leave it.

File: psh/test-cat.py [hunk 2]
```diff
@@ -27,8 +25,61 @@ def assert_cat_h(p):
...
+
+
+def assert_cat_basic(p):
+    fname = ROOT_TEST_DIR + "/textFile.txt"
+    secFname = ROOT_TEST_DIR + "/textFile2.txt"
+    fspecial = ROOT_TEST_DIR + "/text!#$?@File.txt"
+
+    text = "Lorem ipsum dolor sit amet, consectetur adipiscing elit. "
+    secText = "Curabitur eu velit a sem porttitor blandit."
+    ascii = "0123456789?@ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz"
```
Comment (damianloew) [inline psh/test-cat.py:45]: Let's check all printable characters using `string.printable`

File: psh/test-cat.py [hunk 3]
```diff
@@ -27,8 +25,61 @@ def assert_cat_h(p):
...
+
+    # Printing content of file
+    psh.assert_cmd(p, f'echo "{text}" > {fname}')
+    psh.assert_cmd(p, f'cat {fname}', expected=text)
+
+    psh.assert_cmd(p, f'echo "{secText}" > {secFname}')
+    psh.assert_cmd(p, f'cat {secFname}', expected=secText)
+
+    # Concrate two text files
+    psh.assert_cmd(p, f'cat {fname} {secFname}', expected=[text, secText])
```
Comment (damianloew) [inline psh/test-cat.py:62]: Let's use tuple if possible.

```suggestion
    psh.assert_cmd(p, f'cat {fname} {secFname}', expected=(text, secText))
```

Comment (damianloew) [reply psh/test-cat.py:62]: Applies to the other places.

File: psh/test-cat.py [hunk 4]
```diff
@@ -27,8 +25,61 @@ def assert_cat_h(p):
...
+
+    # Print basic characters
+    psh.assert_cmd(p, f'echo "{ascii}" > {secFname}')
+    psh.assert_cmd(p, f'cat {secFname}', expected=ascii)
+
+    # Concrate existing file with non existing
+    statement = 'cat: nonexistentFile: ENOENT'
```
Comment (damianloew) [inline psh/test-cat.py:69]: Please use a regex to be more flexible.

**Review by damianloew — CHANGES_REQUESTED** [review]
Thanks or those improvements! Please be more specific in commit messages if possible.

File: psh/test-cat.py (hunk 2 above)
Comment (mateusz-bloch) [reply psh/test-cat.py]: So, yes, it is possible to use a larger number of characters. I have now added more. 

In theory, by using the echo command, we can write them all thanks to the `-e` argument in echo. However, it is not available on Phoenix.

Comment (damianloew) [reply psh/test-cat.py]: how about whitespaces? We can also test writing them.

File: psh/test.yaml [hunk 5]
```diff
@@ -52,14 +52,10 @@ test:
           harness: test-mkdir.py
 
         - name: touch
-          # TODO touch of netsocket and pf fails...
           harness: test-touch.py
-          ignore: True
 
         - name: touch-rootfs
           harness: test-touch-rootfs.py
-          # temporary disabled because of the following issue: https://github.com/phoenix-rtos/phoenix-rtos-project/issues/621
```
Comment (damianloew) [inline psh/test.yaml:61]: Please remove it as it's done in the another PR.

**Review by damianloew — CHANGES_REQUESTED** [review]
I'd suggest splitting those changes to the following commits:

 - `psh: cat test improvements` -> `psh: test-cat: add cases with writing a content`

- `psh: add cleanup in tests`

- `psh: enable back test-touch`

- `psh: small improvements and cleanup in tests`


IMHO it will be much more readable in commit messages what have been changed.

**Comment by damianloew** [comment]
> I'd suggest splitting those changes to the following commits:
> 
> * `psh: cat test improvements` -> `psh: test-cat: add cases with writing a content`
> * `psh: add cleanup in tests`
> * `psh: enable back test-touch`
> * `psh: small improvements and cleanup in tests`
> 
> IMHO it will be much more readable in commit messages what have been changed.

Of course you can change the messages a little bit if you want

**Review by damianloew — APPROVED** [review]
LGTM
